diff -u -r aenet-2.0.3/src/ext/feedforward.f90 aenet/src/ext/feedforward.f90
--- aenet-2.0.3/src/ext/feedforward.f90	2018-06-25 23:50:01.000000000 +0900
+++ aenet/src/ext/feedforward.f90	2020-05-29 15:52:17.875726683 +0900
@@ -429,6 +429,10 @@
           fname = 'scaled hyperbolic tangent (mtanh)'
        case(4)
           fname = 'scaled hyperbolic tangent + linear twisting (twist)'
+       case(5)
+          fname = 'sigmoid-weighted linear units (swish)'
+       case(6)
+          fname = 'A Self Regularized Non-Monotonic Neural Activation Function (Mish)'
        end select
        write(*,'(5x,I3," : ",I5,2x,A)') ilayer, net%nnodes(ilayer), trim(fname)
     end do
@@ -586,7 +590,7 @@
     integer,                                  intent(in)    :: ny
     double precision, dimension(net%nvalues), intent(out)   :: values
     double precision, dimension(net%nvalues), intent(out)   :: derivs
-    double precision, dimension(nx),          intent(out)   :: y
+    double precision, dimension(ny),          intent(out)   :: y
 
     integer, dimension(2) :: Wshape
     integer               :: iw1, iw2
@@ -896,6 +900,10 @@
        net%f_a(ilayer) = 3
     case('twist','w')
        net%f_a(ilayer) = 4
+    case('swish','i')
+       net%f_a(ilayer) = 5
+    case('mish','h')
+       net%f_a(ilayer) = 6
     case default
        write(0,*) "Error: invalid function type in `change_activation': ", &
                   trim(ftype)
@@ -935,6 +943,11 @@
     double precision, parameter :: a = 1.7159d0
     double precision, parameter :: b = 0.666666666666667d0
     double precision, parameter :: c = 0.1d0
+    double precision :: sigmoid_bx
+    double precision, parameter :: beta = 1.0d0
+    double precision :: softplus_x
+    double precision :: delta,omega
+    
 
     select case(t)
     case(0)
@@ -954,6 +967,16 @@
        tanhbx = tanh(b*x)
        y  = a*tanhbx + c*x
        dy = a*(1.0d0 - tanhbx*tanhbx)*b + c
+    case(5)
+       sigmoid_bx  = (1.0d0/(1.0d0 + exp(-beta*x)))
+       y  = x*sigmoid_bx
+       dy = beta*y + sigmoid_bx*(1.d0 - beta*y)
+    case(6)
+       softplus_x = log(1+exp(x))
+       y  = x*tanh(softplus_x)
+       omega = 4*(x+1) + 4*exp(2*x) + exp(3*x) + exp(x)*(4*x+6)
+       delta = 2*exp(x) + exp(2*x) + 2
+       dy = exp(x)*omega/delta**2
     case default
        y  = 0.0d0
        dy = 0.0d0
diff -u -r aenet-2.0.3/src/ext/sfbasis.f90 aenet/src/ext/sfbasis.f90
--- aenet-2.0.3/src/ext/sfbasis.f90	2018-06-25 23:50:01.000000000 +0900
+++ aenet/src/ext/sfbasis.f90	2020-10-29 14:30:36.275800459 +0900
@@ -19,6 +19,9 @@
 
 module sfbasis
 
+  use, intrinsic :: iso_c_binding, only: c_double, c_int, c_char, c_bool, &
+                                         c_ptr, c_f_pointer, c_null_char
+
   use io, only: io_unit
 
   use chebyshev, only: chebyshev_polynomial, &
@@ -66,6 +69,14 @@
   double precision, parameter, private :: PI_INV = 1.0d0/PI
   double precision, parameter, private :: PI2    = 2.0d0*PI
   double precision, parameter, private :: EPS    = 1.0d-12
+  
+  logical                              :: switch_sfb = .false.
+  integer                              :: tcof_r = 0
+  integer                              :: tcof_a = 0
+  double precision                     :: r_h = 1.5d0
+  double precision                     :: a_h = 1.5d0
+  
+  integer(kind=c_int), bind(C, name='aenet_sfb_ver'), public :: aenet_sfb_ver = 3_c_int
 
 contains
 
@@ -139,6 +150,29 @@
        s = s + 1
     end do
 
+    select case(aenet_sfb_ver)
+    case(1)
+       switch_sfb = .true.
+       tcof_r  = 0
+       tcof_a  = 0
+    case(2)
+       switch_sfb = .true.
+       tcof_r = 1
+       tcof_a = 1
+       r_h = 3.d0
+       a_h = 5.d0
+    case(3)
+       switch_sfb = .true.
+       tcof_r = 1
+       tcof_a = 2
+       r_h = 3.d0
+       a_h = 1.25d0
+    case default
+       switch_sfb = .false.
+       tcof_r  = 0
+       tcof_a  = 0
+    end select
+    
     sfb%initialized = .true.
 
   end function new_SFBasis
@@ -709,37 +743,59 @@
   !====================================================================!
 
 
-  function sfb_fc(Rij, Rc) result(fc)
+  function sfb_fc(Rij, Rc, tcof, h) result(fc)
 
     implicit none
 
-    double precision, intent(in) :: Rij, Rc
+    double precision, intent(in) :: Rij, Rc, h
+    integer,intent(in)           :: tcof
     double precision             :: fc
+    
+    double precision :: x
 
     if (Rij >= Rc) then
        fc  = 0.0d0
     else
-       fc  =  0.5d0*(cos(PI/Rc*Rij) + 1.0d0)
+       select case (tcof)
+       case(0)
+          fc  =  0.5d0*(cos(PI/Rc*Rij) + 1.0d0)
+       case(1)
+          fc  = 1.d0 + h*(Rij/Rc)**(h+1) - (h+1)*(Rij/Rc)**(h)
+       case(2)
+          x  = (Rij - Rc)/h
+          fc  = x**2/(1+x**2)
+       end select
     end if
 
   end function sfb_fc
 
   !--------------------------------------------------------------------!
 
-  function sfb_fc_d1(Rij, Rc) result(dfc)
+  function sfb_fc_d1(Rij, Rc, tcof, h) result(dfc)
 
     implicit none
 
-    double precision, intent(in) :: Rij, Rc
+    double precision, intent(in) :: Rij, Rc, h
+    integer,intent(in)           :: tcof
     double precision             :: dfc
-
+    
+    double precision :: x
     double precision :: a
 
     if (Rij >= Rc) then
        dfc = 0.0d0
     else
-       a = PI/Rc
-       dfc = -0.5d0*a*sin(a*Rij)
+       select case (tcof)
+       case(0)
+          a = PI/Rc
+          dfc = -0.5d0*a*sin(a*Rij)
+       case(1)
+          dfc = (h*(h+1)*(Rij/Rc)**(h) - (h+1)*h*(Rij/Rc)**(h-1))/Rc
+       case(2)
+          x  = (Rij - Rc)/h
+          dfc = 2*x/(1+x**2)**2/h
+       end select
+       
     end if
 
   end function sfb_fc_d1
@@ -769,13 +825,13 @@
 
     call sfb_assert_init(sfb)
 
-    w_ij = sfb_fc(d_ij, sfb%r_Rc)
+    w_ij = sfb_fc(d_ij, sfb%r_Rc,tcof_r,r_h)
     f = chebyshev_polynomial(d_ij, 0.0d0, sfb%r_Rc, sfb%r_order)
 
     values(1:sfb%r_N) = w_ij*f(1:sfb%r_N)
 
     if (present(deriv_i) .and. present(deriv_j)) then
-       dw_ij = sfb_fc_d1(d_ij, sfb%r_Rc)
+       dw_ij = sfb_fc_d1(d_ij, sfb%r_Rc, tcof_r, r_h)          
        df = chebyshev_polynomial_d1(d_ij, 0.0d0, sfb%r_Rc, sfb%r_order)
        forall (i=1:sfb%r_N)
           deriv_i(:,i) = -R_ij/d_ij*(dw_ij*f(i) + w_ij*df(i))
@@ -808,20 +864,32 @@
     double precision, dimension(3)       :: di_cos_ikj, dj_cos_ikj, dk_cos_ikj
     double precision, dimension(3)       :: di_w_ijk, dj_w_ijk, dk_w_ijk
     integer                              :: i
+    
+    double precision                     :: h_cos
 
     call sfb_assert_init(sfb)
-
-    fc_j = sfb_fc(d_ij, sfb%a_Rc)
-    fc_k = sfb_fc(d_ik, sfb%a_Rc)
-    w_ijk = fc_j*fc_k
-    f = chebyshev_polynomial(cos_ijk, 0.0d0, PI, sfb%a_order)
+    
+    fc_j = sfb_fc(d_ij, sfb%a_Rc, tcof_a, a_h)
+    fc_k = sfb_fc(d_ik, sfb%a_Rc, tcof_a, a_h)
+    w_ijk = fc_j*fc_k    
+    if(switch_sfb)then
+       h_cos = 0.5d0*(cos_ijk+1.d0)
+       f = chebyshev_polynomial(h_cos, -1.d0, 1.d0, sfb%a_order)
+    else
+       f = chebyshev_polynomial(cos_ijk, 0.0d0, PI, sfb%a_order)
+    end if
 
     values(1:sfb%a_N) = w_ijk*f
 
     if (present(deriv_i) .and. present(deriv_j) .and. present(deriv_k)) then
-       dfc_j = sfb_fc_d1(d_ij, sfb%a_Rc)
-       dfc_k = sfb_fc_d1(d_ik, sfb%a_Rc)
-       df = chebyshev_polynomial_d1(cos_ijk, 0.0d0, PI, sfb%a_order)
+       dfc_j = sfb_fc_d1(d_ij, sfb%a_Rc, tcof_a, a_h)
+       dfc_k = sfb_fc_d1(d_ik, sfb%a_Rc, tcof_a, a_h)
+       if(switch_sfb)then
+          df = 0.5d0*chebyshev_polynomial_d1(h_cos, -1.0d0, 1.d0, sfb%a_order)
+       else
+          df = chebyshev_polynomial_d1(cos_ijk, 0.0d0, PI, sfb%a_order)
+       end if
+       
        id_ij2 = 1.0d0/(d_ij*d_ij)
        id_ik2 = 1.0d0/(d_ik*d_ik)
        id_ij_ik = 1.0d0/(d_ij*d_ik)
diff -u -r aenet-2.0.3/src/Makefile.inc aenet/src/Makefile.inc
--- aenet-2.0.3/src/Makefile.inc	2018-06-25 23:50:01.000000000 +0900
+++ aenet/src/Makefile.inc	2020-05-29 15:52:17.875726683 +0900
@@ -57,7 +57,7 @@
 generate : $(TARGET1)
 train    : $(TARGET2)
 predict  : $(TARGET3)
-lib      : libaenet.a libaenet.so
+lib      : ../../libaenet.a libaenet.so
 
 %.o : %.f90
 	$(FC) $(FCFLAGS) $< -o $*.o
@@ -83,15 +83,16 @@
 $(BIN)%.x : %.F90 $(OBJECTS) Makefile.inc
 	$(LD) $(LDFLAGS) -o $@-$(version)-$(SUFFIX) $< $(OBJECTS) $(LIB) $(NUMLIB)
 
-libaenet.a : $(OBJECTS) Makefile.inc
+../../libaenet.a : $(OBJECTS) Makefile.inc
 	$(AR) $(ARFLAGS) $@ $(OBJECTS)
+	@cp $(EXTRAMAKE) ../../Makefile.lammps
 
 libaenet.so : $(PICOBJECTS) Makefile.inc
 	$(LIBLD) $(LIBFLAGS) $(PICOBJECTS) $(LIBLIB) -o $@
 
 veryclean: clean
 	for f in $(TARGETS); do rm -f $$f-*; done
-	if [ -e libaenet.a ]; then rm -f libaenet.a; fi
+	if [ -e ../../libaenet.a ]; then rm -f ../../libaenet.a; fi
 	if [ -e libaenet.so ]; then rm -f libaenet.so; fi
 	if [ -e TAGS ]; then rm -f TAGS; fi
 	for f in $(TESTS); do rm -f $$f-*; done
diff -u -r aenet-2.0.3/src/makefiles/Makefile.gfortran_serial aenet/src/makefiles/Makefile.gfortran_serial
--- aenet-2.0.3/src/makefiles/Makefile.gfortran_serial	2018-06-25 23:50:01.000000000 +0900
+++ aenet/src/makefiles/Makefile.gfortran_serial	2020-05-29 15:52:17.879726683 +0900
@@ -18,12 +18,14 @@
 # 2015-05-26 Alexander Urban (AU) and Nongnuch Artrith (NA)
 #-----------------------------------------------------------------------
 
+EXTRAMAKE = ../../Makefile.lammps.gfortran
+
 SUFFIX   = gfortran_serial
 
 FC       = gfortran -c
 LD       = gfortran
 DEBUG    =
-FCFLAGS  = -O3 -pedantic -fexternal-blas $(DEBUG)
+FCFLAGS  = -O3 -pedantic $(DEBUG)
 LDFLAGS  = -static-libgfortran $(DEBUG)
 NUMLIB   = -llapack -lblas
 
diff -u -r aenet-2.0.3/src/makefiles/Makefile.ifort_serial aenet/src/makefiles/Makefile.ifort_serial
--- aenet-2.0.3/src/makefiles/Makefile.ifort_serial	2018-06-25 23:50:01.000000000 +0900
+++ aenet/src/makefiles/Makefile.ifort_serial	2020-05-29 15:52:17.879726683 +0900
@@ -18,11 +18,13 @@
 # 2015-05-26 Alexander Urban (AU) and Nongnuch Artrith (NA)
 #-----------------------------------------------------------------------
 
+EXTRAMAKE = ../../Makefile.lammps.ifort
+
 SUFFIX   = ifort_serial
 
 FC       = ifort -c
 LD       = ifort
-DEBUG    = -g -warn all -check bounds
+DEBUG    = 
 FCFLAGS  = -O2 $(DEBUG)
 LDFLAGS  = -static-intel $(DEBUG)
 NUMLIB   = -mkl
